base/libgit2/utils.jl

```diff
@@ -3,7 +3,7 @@
 # Parse "GIT URLs" syntax (URLs and a scp-like syntax). For details see:
 # https://git-scm.com/docs/git-clone#_git_urls_a_id_urls_a
 const URL_REGEX = r"""
-^(?:(?<scheme>ssh|git|https?)://)?
+^(?:(?<scheme>ssh|git|https?)://)?+
```

Comment:
Basically when `https://` is matched the regular expression will stop any backtracking where this string could be repurposed elsewhere in the regular expression:

```julia
julia> match(possessive_regex, "https://git@server:repo")

julia> match(original_regex, "https://git@server:repo")
RegexMatch("https://git@server:repo", scheme=nothing, user="https", password="//git", host="server", port=nothing, path="repo")
```